### PR TITLE
feat: adds a busy spinner to combobox even when caret is hidden

### PR DESCRIPTION
**Pull request by brian-bfly**
image

### CODE REVIEW

File: packages/react-widgets/src/Combobox.tsx [hunk 1]
```diff
@@ -405,9 +406,32 @@ const ComboboxImpl: Combobox = React.forwardRef(function Combobox<TDataItem>(
 
   let popupOpen = currentOpen && (!hideEmptyPopup || !!data.length)
   let inputReadOnly =
-  // @ts-ignore
-      inputProps?.readOnly != null ? inputProps?.readOnly : readOnly;
-      
+    // @ts-ignore
+    inputProps?.readOnly != null ? inputProps?.readOnly : readOnly
+
+  const Addon = () => {
```
[the remark this answers is not in the document]
Comment (brian-bfly) [reply packages/react-widgets/src/Combobox.tsx]: Done.

File: www/plugins/examples/busy.js [hunk 2]
```diff
@@ -14,6 +14,8 @@ module.exports = function (widgetName) {
     <>
       <${widgetName} busy />
       <br/>
+      <${widgetName} busy hideCaret />
```
Comment (jquense) [inline www/plugins/examples/busy.js:17]: I believe this example is used by multiple components, not all of which have "hideCaret" as a prop. We'd want to make a combobox specific one, or probably better make this part of a `hideCaret` example instead.

**Comment by jquense** [comment]
image

Notice that the input box shadow isn't extending underneath the caret. This is because the input has the shadow and is sharing space with the caret box. There are probably a few ways to address this, but probably the easiest is to add the `rw-widget-input` to the container div when `hideCaret` is true so the box shadow is around everything

File: packages/react-widgets/src/Combobox.tsx [hunk 3]
```diff
@@ -430,7 +453,7 @@ const ComboboxImpl: Combobox = React.forwardRef(function Combobox<TDataItem>(
           className={cn(
             // @ts-ignore
             inputProps && inputProps.className,
-            'rw-widget-input rw-combobox-input',
+            'rw-combobox-input',
```
[the remark this answers is not in the document]
Comment (brian-bfly) [reply packages/react-widgets/src/Combobox.tsx:460]: Done.

File: packages/react-widgets/src/WidgetPicker.tsx [hunk 4]
```diff
@@ -4,16 +4,17 @@ import React from 'react'
...
+}: React.HTMLProps<HTMLDivElement> & { hideCaret?: boolean, busy?: boolean }) {
   return (
     <div
       {...props}
       className={cn(
         className,
         'rw-widget-picker',
         'rw-widget-container',
-        hideCaret && 'rw-hide-caret',
+        (hideCaret && !busy) && 'rw-hide-caret',
```
Comment (jquense) [inline packages/react-widgets/src/WidgetPicker.tsx:17]: last thing!, i wouldn't conflate the meaning here as this isn't what you'd expect in other components. We can do this logic in Combobox where we are synthesizing the hide meaning more specifically

File: www/plugins/examples/busy.js (hunk 2 above)
Comment (jquense) [reply www/plugins/examples/busy.js:17]: revert this?

Comment (brian-bfly) [reply www/plugins/examples/busy.js]: Done.

**Review by jquense — APPROVED** [review]

**Pull request merged**